lib/std/Build/Step/Compile.zig

```diff
@@ -952,6 +953,17 @@ pub fn addWin32ResourceFile(self: *Compile, source: RcSourceFile) void {
     source.file.addStepDependencies(&self.step);
 }
 
+pub fn addWin32ManifestFile(self: *Compile, source: LazyPath) void {
```

Comment:
I could be convinced either way, but did you consider adding a `win32_manifest` field to `std.Build.ExecutableOptions` rather than this since there can only be possibly one manifest file?

Reply:
I think that would be better. It's actually what I had in mind originally but I guess I got carried away copy+pasting from the resource file stuff.

Reply:
Ended up going with a `win32_manifest` field but with a doc comment saying not to modify it directly and use `setWin32ManifestFile` instead, since the function is able to setup the step dependency for the LazyPath. Let me know if that makes sense to you.

Reply:
I think it might be better to follow these examples:

https://github.com/ziglang/zig/blob/a1e0b9979ab2a47ad1d13dd06106c3143d9a1c9e/lib/std/Build/Step/Compile.zig#L518-L526

Generally, I think it will be better to have fewer state mutations in the API.

Reply:
My understanding is that those only can work that way because they are part of the `Options` given to `create`. I couldn't find an example of a field that works that way outside of `Options`.

Would you like me to add `win32_manifest` to the `Options` struct? Or is there some other place that the `addStepDependencies` call could happen?

Reply:
Yes that is my suggestion - adding the field to the `Options` struct. This maintains the property that whether or not a manifest file will be part of the compilation is known for the entire lifetime of the Compile step. This may not be strictly necessary, but you can imagine that having this property for almost all of the state makes things simpler to reason about.

Reply:
Note that the field was intentionally not added to `ObjectOptions` or `StaticLibraryOptions`:

- When building object files, resources just get lost entirely (they don't exist anywhere in the `.obj` file)
- When building static libraries, resources do get put into the `.lib` but when linking the resources in the static library get lost (https://stackoverflow.com/questions/531502/vc-resources-in-a-static-library)

(I'll think more about this with regards to `.rc` files, since the same caveats apply to them)